### PR TITLE
PR2 for BXMSDOC-7217: Release notes docs for DM and PAM 7.10

### PR DESCRIPTION
**Pull request by kaldesai**
**JIRAs:** 

- [Epic](https://issues.redhat.com/browse/BXMSDOC-6835)

**Doc previews:** 
- [Release notes for Red Hat Process Automation Manager 7.10](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-7217-RN-7.10-RHPAM/)
- [Release notes for Red Hat Decision Manager 7.10](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-7217-RN-7.10-RHDM/)

### CODE REVIEW

File: doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc [hunk 1]
```diff
@@ -5,180 +5,182 @@ This section highlights new features in {PRODUCT} {PRODUCT_VERSION}.
...
-=== Process designer updates
-The following list provides a summary of process designer updates:
+{PRODUCT} now provides standalone editors for Business Process Model and Notation (BPMN) process models and Decision Model and Notation (DMN) decision models. The standalone editors enable you to view and design BPMN and DMN models embedded in your web applications. The read-only mode for standalone BPMN editor is not supported in {PRODUCT} 7.10.0.
 
-* Ability to add data objects in a process diagram
-* Ability to add Kafka work item in a process diagram to send events to a Kafka topic
-* Ability to double-click any diagram node to edit its name using the new inline text editor
-* Ability to view highlighted syntax when writing a script in Java, JavaScript, and MVEL
-* Ability to use multiple connections for a task or an event using `jbpm.enable.multi.con` system property
+The standalone editors are distributed in a self-contained library that provides an all-in-one JavaScript file for each editor. The JavaScript file uses a comprehensive API to set and control the editor. You can install the standalone editors in three ways:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc:64]:
```suggestion
The standalone editors are distributed in a self-contained library that provides an all-in-one JavaScript file for each editor. The JavaScript file uses a comprehensive API to set and control the editor. You can install the standalone editors in two ways:
```

File: doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc [hunk 2]
```diff
@@ -5,180 +5,182 @@ This section highlights new features in {PRODUCT} {PRODUCT_VERSION}.
...
-You can follow the example in {URL_DEVELOPING_SOLVERS}#assembly-business-optimizer-springboot[{PLANNER_WITH_SPRING_BOOT}] to create a school timetable.
+The following list provides a summary of the process engine API updates:
 
-=== Employee Rostering rotation pattern redesign
+* You can now use the {PROCESS_ENGINE} API to signal a process instance using its correlation key.
+* You can now use the {PROCESS_ENGINE} API to determine the node type from within an event listener.
+* When using the advanced query feature of the {PROCESS_ENGINE} API, you can now optionally exclude process variables from the search results.
 
-The rotation pattern on the Employee Rostering Rotation window has been redesigned for ease of use. For more information, see {URL_DEVELOPING_SOLVERS}#assembly-optimizer-modifying-ER-template-IDE[_{RUNNING_ROSTER_FOR_PLANNER_IDE}_].
+For more information about using the {PROCESS_ENGINE} API, see {URL_DEPLOYING_AND_MANAGING_SERVICES}#assembly-kie-apis[_{KIE_APIS}_].
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc:117]: AFAIK, the updates related to process engine API have not been covered in KIE API document. Can you please confirm this with Misha once?

File: doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc [hunk 3]
```diff
@@ -5,180 +5,182 @@ This section highlights new features in {PRODUCT} {PRODUCT_VERSION}.
...
 
-== Spring Boot
+=== KIE API improvements
 
-ifdef::PAM[]
+You can now use the KIE REST API to signal a process using its alias. For more information about using the KIE REST API, see {URL_DEPLOYING_AND_MANAGING_SERVICES}#assembly-kie-apis[_{KIE_APIS}_].
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc:121]: Please confirm with Misha about the link

File: doc-content/enterprise-only/release-notes/rn-7.10-fixed-issues-ref.adoc [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+* REST endpoints are accessible to all the logged in users, even when the users are assigned without the correct roles [https://issues.redhat.com/browse/RHPAM-3284[RHPAM-3284]]
+* In test scenarios coverage report, rule name column size is increased and the numbers are aligned to the right side [https://issues.redhat.com/browse/RHPAM-3271[RHPAM-3271]]
+* In a guided decision table, an exception error pop-up returns an inappropriate error message [https://issues.redhat.com/browse/RHPAM-3201[RHPAM-3201]]
+* The number of assets listed on the project page of {CENTRAL} does not match with the count displayed for pagination [https://issues.redhat.com/browse/RHPAM-2859[RHPAM-2859]]
+
+ifdef::PAM[]
+
+== Form modeler
+
+* `DocumentCollection` output parameter does not work in generated forms [https://issues.redhat.com/browse/RHPAM-3252[RHPAM-3252]]
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/release-notes/rn-7.10-fixed-issues-ref.adoc:19]:
```suggestion
* `DocumentCollection` output parameter does not work in the generated forms [https://issues.redhat.com/browse/RHPAM-3252[RHPAM-3252]]
```

File: doc-content/enterprise-only/release-notes/rn-7.10-fixed-issues-ref.adoc [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+endif::[]
+
+== {KIE_SERVER}
+
+ifdef::PAM[]
+
+* Some issues appear when starting up the {KIE_SERVER} while executing a long query [https://issues.redhat.com/browse/RHPAM-3342[RHPAM-3342]]
+* When a specified time is reached, a `start timer` event in a process model creates duplicate process instances [https://issues.redhat.com/browse/RHPAM-3189[RHPAM-3189]]
+* When you generate a form with a components slider, various form options including DocumentCollection, MultipleInput, and MultipleSelector, are not rendering in an API response [https://issues.redhat.com/browse/RHPAM-2330[RHPAM-2330]]
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/release-notes/rn-7.10-fixed-issues-ref.adoc:29]:
```suggestion
* When you generate a form with a components slider, various form options including `DocumentCollection`, `MultipleInput`, and `MultipleSelector`, are not rendering in an API response [https://issues.redhat.com/browse/RHPAM-2330[RHPAM-2330]]
```

File: doc-content/enterprise-only/release-notes/rn-7.10-fixed-issues-ref.adoc [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+== DMN designer
+
+* DMN editor does not detect the changes from the Decision Navigator view [https://issues.redhat.com/browse/RHDM-1482[RHDM-1482]]
+
+ifdef::PAM[]
+
+== Process engine
+
+* You cannot use the same `correlationKey` for two active process instances [https://issues.redhat.com/browse/RHPAM-3336[RHPAM-3336]]
+* When you set `userId == null`, `exlOwner !=null` and, `groups !=null` in a query, the task is not returned [https://issues.redhat.com/browse/RHPAM-3230[RHPAM-3230]]
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/release-notes/rn-7.10-fixed-issues-ref.adoc:44]:
```suggestion
* When you set `userId == null`, `exlOwner !=null`, and `groups !=null` in a query, the task is not returned [https://issues.redhat.com/browse/RHPAM-3230[RHPAM-3230]]
```

File: doc-content/enterprise-only/release-notes/rn-7.10-fixed-issues-ref.adoc [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+* In MVEL dialect mode in an executable model, when you use an enhanced `for` (foreach) statement, you receive a compilation error [https://issues.redhat.com/browse/RHDM-1513[RHDM-1513]]
+* In MVEL dialect mode in an executable model, when you use the space characters between `(` and `LocalVariableType` in an enhanced `for` (foreach) statement, you receive a compilation error [https://issues.redhat.com/browse/RHDM-1505[RHDM-1505]]
+* When you use the standard Java serialization mechanism, the `ByteArrayResource` is declared twice [https://issues.redhat.com/browse/RHDM-1502[RHDM-1502]]
+* When you are unable to call a `reset` method in the `PriorityMap` and call the method inside `org.kie.api.internal.utils.ServiceDiscoveryImpl`, you receive an exception [https://issues.redhat.com/browse/RHDM-1501[RHDM-1501]]
+* When you have two or more different implementations of service on the classpath, the service discovery or wiring mechanism randomly chooses the services that come first in the classpath. You can now explicitly configure a priority among these services [https://issues.redhat.com/browse/RHDM-1500[RHDM-1500]]
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/release-notes/rn-7.10-fixed-issues-ref.adoc:66]:
```suggestion
* When you have two or more different implementations of service on the classpath, the service discovery or wiring mechanism randomly chooses the services that come first in the classpath. You can explicitly configure a priority among these services [https://issues.redhat.com/browse/RHDM-1500[RHDM-1500]]
```

File: doc-content/enterprise-only/release-notes/rn-7.10-known-issues-ref.adoc [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+Expected result: No errors occur in the user interface or server log.
+
+Actual result: You receive an unexpected system error.
+
+Workaround: None.
+
+.Service task thumbnails are large in documentation [https://issues.redhat.com/browse/RHPAM-2759[RHPAM-2759]]
+
+Issue: Service task thumbnails are large as compared to other nodes thumbnails in *Documentation*.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/release-notes/rn-7.10-known-issues-ref.adoc:101]:
```suggestion
Issue: Service task thumbnails are large as compared to other thumbnails in *Documentation*.
```

File: doc-content/enterprise-only/release-notes/rn-7.10-known-issues-ref.adoc [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
++
+The generated BPMN file does not contain the `structureRef` for `endMessageType`, as per the following example:
++
+[source]
+----
+<bpmn2:itemDefinition id="endMessageType"/>
+<bpmn2:message id="_Icr1JGVqEeuMVv5wG-FCkw" itemRef="endMessageType" name="endMessage"/>
+----
+
+Workaround: Perform the same steps in the VSCode editor in desktop, the correct BPMN file is generated, as per the following example:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/release-notes/rn-7.10-known-issues-ref.adoc:136]:
```suggestion
Workaround: Perform the same steps in the VSCode editor in desktop, the correct BPMN file is generated as per the following example:
```

**Review by hmanwani-rh — COMMENTED** [review]
@kaldesai Please check the few minor comments below and let me know if you have any questions.

File: doc-content/enterprise-only/release-notes/rn-7.10-fixed-issues-ref.adoc [hunk 10]
```diff
@@ -0,0 +1,75 @@
+[id='rn-7.10-fixed-issues-ref']
+= Fixed issues in {PRODUCT} {ENTERPRISE_VERSION_LONG}
+
+{PRODUCT} {ENTERPRISE_VERSION_LONG} provides increased stability and fixed issues listed in this section.
+
+== {CENTRAL}
+
+* Package list fails to populate default package for imported projects [https://issues.redhat.com/browse/RHPAM-3209[RHPAM-3209]]
+* Collection editor is not opening for DRL based test scenarios [https://issues.redhat.com/browse/RHDM-1464[RHDM-1464]]
+* REST endpoints are accessible to all the logged in users, even when the users are assigned without the correct roles [https://issues.redhat.com/browse/RHPAM-3284[RHPAM-3284]]
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/enterprise-only/release-notes/rn-7.10-fixed-issues-ref.adoc:10]: Done

File: doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc [hunk 11]
```diff
@@ -5,180 +5,182 @@ This section highlights new features in {PRODUCT} {PRODUCT_VERSION}.
...
-=== Support for multiple diagrams in DMN models
+=== Support for DMN 1.1 and 1.3 assets in DMN designer
 
-For complex DMN models, you can now use the DMN designer in {CENTRAL} to design multiple DMN decision requirements diagrams (DRDs) that represent parts of the overall decision requirements graph (DRG) for a DMN decision model. This improvement enables you to better organize DMN decision models with many decision requirements by dividing the model into smaller nested DRDs.
+You can now use the DMN designer in {CENTRAL} to open DMN 1.1 and 1.3 assets. Any DMN 1.1 and 1.3 models (do not contain DMN 1.3 features) that you import into {CENTRAL}, open in the DMN designer, and save are converted to DMN 1.2 models. For more information, see {URL_DEVELOPING_DECISION_SERVICES}#assembly-dmn-models[_{DMN_MODELS}_].
 
-For more information about creating DMN models with multiple diagrams, see {URL_DEVELOPING_DECISION_SERVICES}#assembly-dmn-models[_{DMN_MODELS}_].
+=== Test scenarios
+
+The test scenarios designer in {CENTRAL} is now available with the new features:
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc:14]: Done

File: doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc [hunk 12]
```diff
@@ -5,180 +5,182 @@ This section highlights new features in {PRODUCT} {PRODUCT_VERSION}.
...
+
+* You can now perform the undo and redo operations from the *Settings* panel. Also, the warning pop-up is indicated when you save or discard any changes from the *Settings* panel.
```
[the remark this answers is not in the document]
Comment (kaldesai) [reply doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc:16]: Done

File: doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc (hunk 2 above)
Comment (kaldesai) [reply doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc:117]: I verified with Misha and his response was:
There was a discussion of these API changes, but on the REST API side, they are covered in the Swagger and not in the doc. They are also covered in the Process Engine doc from the Java API side So he said I have added those links just for more information about APIs and nothing else.

File: doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc (hunk 3 above)
Comment (kaldesai) [reply doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc:121]: Same as above

File: doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc [hunk 13]
```diff
@@ -5,180 +5,182 @@ This section highlights new features in {PRODUCT} {PRODUCT_VERSION}.
...
+* Dashbuilder Runtime is now available with a new heatmap component feature. You can add the heatmap component to the pages and export it to the Dashbuilder Runtime. Heatmap components are used to display heat information over a process diagram, and you can create, edit, and build a dashboard using the heatmap component. The heat information is retrieved from {KIE_SERVER} datasets.
++
+For more information, see {URL_MANAGING_SETTINGS}#assembly-building-custom-dashboard-widgets[_{BUILDING_WIDGETS}_].
 
 endif::PAM[]
 
 ifdef::PAM[]
 
-== Process Designer
+== New standalone BPMN and DMN editors
```
Comment (krisv) [inline doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc:60]: Shouldn't the standalone DMN editor be described for RHDM as well?

Comment (kaldesai) [reply doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc:60]: @krisv, I received the feedback from Martin Weiler that **New standalone BPMN and DMN editors** feature not applicable to RHDM so I removed from RHDM doc. In that case, I will include it in the RHDM doc as well. 
Can you please confirm?

Comment (krisv) [reply doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc:60]: @kaldesai afaik Martin was wondering whether the BPMN editor should be mentioned for RHDM (I think there was already agreement that at least the DMN editor should be mentioned).  But BPMN is also used for RuleFlow in RHDM, so I don't think there's an issue with mentioning it in RHDM as well, and Ivo confirmed in his JIRA comment as well.

Comment (kaldesai) [reply doc-content/enterprise-only/release-notes/rn-whats-new-con.adoc:60]: @krisv, Okay thank you for the confirmation I will include it in the RHDM doc.

File: doc-content/enterprise-only/release-notes/rn-kogito-productize-features-ref.adoc [hunk 14]
```diff
@@ -0,0 +1,17 @@
+[id='rn-kogito-productize-features-ref']
+= Kogito productized features in {PRODUCT}
+
+{PRODUCT} {ENTERPRISE_VERSION} provides the following Kogito based applications that you can use to design Business Process Model and Notation (BPMN) process models and Decision Model and Notation (DMN) decision models using graphical modelers:
```
Comment (porcelli) [inline doc-content/enterprise-only/release-notes/rn-kogito-productize-features-ref.adoc:4]: We need a way to express that the VS Code tool is not only for Kogito, but also for regular PAM/DM projects.

Those editors were developed under Kogito initiativbe, but are useful for PAM and Kogito.

**Review by porcelli — CHANGES_REQUESTED** [review]

File: doc-content/enterprise-only/release-notes/rn-kogito-productize-features-ref.adoc (hunk 14 above)
Comment (kaldesai) [reply doc-content/enterprise-only/release-notes/rn-kogito-productize-features-ref.adoc:4]: Hey @porcelli, Apart from **Kogito productized features** section I have added a separate **3.3. Support for VSCode extension** section in the release notes and all the details are included in that section.

Comment (porcelli) [reply doc-content/enterprise-only/release-notes/rn-kogito-productize-features-ref.adoc:4]: I understand, but by a user that reads the above sentence would be confuse on what exactly VS Code extensions are expected to support. I'd suggest some better clarification.

Comment (kaldesai) [reply doc-content/enterprise-only/release-notes/rn-kogito-productize-features-ref.adoc:4]: @porcelli, Apart from Kogito productized features section, I have added a separate **3.3. Support for VSCode extension section** in the same release notes and all the details are included in that section.

**Pull request merged**